modules/adapters/java.py

```diff
@@ -0,0 +1,145 @@
+from ..typecheck import Optional, Dict, Any, Tuple
+from ..import core
+from ..import dap
+from .import util
+
+import sublime
+import sublime_plugin
+import os
+import json
+
+
+class Java(dap.AdapterConfiguration):
+	jdtls_bridge: Dict[int, core.Future] = {}
+	jdtls_bridge_current_id = 0
+
+	type = 'java'
+	docs = 'https://github.com/redhat-developer/vscode-java/blob/master/README.md'
+
+	installer = util.OpenVsxInstaller(
+		type='java',
+		repo='vscjava/vscode-java-debug'
+	)
+
+	async def start(self, log, configuration):
+		# Make sure LSP and LSP-JDTLS are installed
+		pc_settings = sublime.load_settings('Package Control.sublime-settings')
+		installed_packages = pc_settings.get('installed_packages', [])
+		if 'LSP-jdtls' not in installed_packages or 'LSP' not in installed_packages:
+			raise core.Error('LSP and LSP-jdtls required to debug Java!')
+
+		# Get configuration from LSP
+		lsp_config = await self.get_configuration_from_lsp()
```

Comment:
I might change that next week to only request if the user did not provide certain config option, by splitting in multiple methods. I should also use the correct main class in the checkProjectSettings command below.